app/controllers/catalog_admin/suggestions_controller.rb

```diff
@@ -0,0 +1,35 @@
+class CatalogAdmin::SuggestionsController < CatalogAdmin::BaseController
+  include ControlsCatalog
+  include ControlsItemList
+
+  before_action :find_item_type
+  before_action :find_item
+  before_action :find_suggestion
+
+  def destroy
+    authorize(@suggestion)
+    @suggestion.destroy
+    redirect_to edit_catalog_admin_item_path(id: @item.id)
+  end
+
+  def update_processed
+    authorize(@suggestion)
+    @suggestion.process
+    redirect_to edit_catalog_admin_item_path(id: @item.id)
+  end
+
+  private
+
+  def find_item_type
+    @item_type =
+      catalog.item_types.where(:slug => params[:item_type_slug]).first!
```

Comment:
```suggestion
    @item_type = catalog.item_types.find_by(slug: params[:item_type_slug])
```
(non testé mais ca devrait passer)

Reply:
yes